### ISSUE TITLE
Project is not posting on Admin panel

### ISSUE THREAD

**Issue by sredXNY** [qor/admin#67]
Hi, I'm developing a project using QOR admin, I setted up a CRUD for a table and it's working great on local enviroment, then I upload the project to a CentOS server and everything seems to be working fine, except that it's not posting (create,edit, delete) the requests but it's listing the registers and showing properly the data.

If I try to create a new register the pages stays loading but it never ends.
I have to say that the database is in another server, it connects but don't know what can be blocking that requests. Any suggestion?  anyone had the same problem?
Thanks!

**Comment by jinzhu** [comment]
Hello @sredXNY 

It should be caused by this middleware. https://github.com/qor/admin/blob/master/route.go#L161-L171

It is used to ensure the POST is not from another website to avoid CRSF

**Comment by jinzhu** [comment]
So you must disabled to include `Referer` in the request header for your browser?

**Comment by sredXNY** [comment]
Hi @jinzhu thanks for your response. I didn't disable anything yet, so which is the proper way to fix it? I don't want to modify the code of qor-admin library, and also don't want to ignore or put aside the security stuffs. 
Thank you!

**Comment by jinzhu** [comment]
What's your browser and OS?

**Comment by sredXNY** [comment]
@jinzhu on Local enviroment Windows8, in the Server CentOS 7, browser: Mozilla Firefox but also tested with Chrome.

I tried overriding the csrf middleware with:

```
Admin.GetRouter().Use(&admin.Middleware{
		Name: "csrf_check",
		Handler: func(context *admin.Context, middleware *admin.Middleware) {
			fmt.Println("pasa por aqui")
			middleware.Next(context)
		},
	})

I put that code just after:
`Admin.MountTo("/admin", mux)`
	
But I get the same result.
```

**Comment by sredXNY** [comment]
To be more specific, we have the database running on the cloud, a server only for the database and other server for the webapp.
In localhost I am using the cloud database, I am pointing to that server and I am able to execute the 4 CRUD operations very well.
Then, when I deployed the application on her own server I am able only to execute the GET petitions, but I cannot create, update or delete. 

So, why I get that CSRF error on the application when is deployed on the server and why not in localhost? also, if I am sending the petitions to the same site why the CSRF validation gave error?

**Comment by jinzhu** [comment]
Are you using any proxy on server? and it dropped the `Referer` header?

**Comment by sredXNY** [comment]
yes, I have installed caddy as HTTP server. Now, I checked and at leat the create action sends the Referer header, please check it http://prntscr.com/f11zii

**Comment by jenshult** [comment]
@sredXNY Maybe you have already handled this issue, but to completely bypass the csrf check, try this: 

`
Admin.GetRouter().GetMiddleware("csrf_check").Handler = func(context *admin.Context, middleware *admin.Middleware)
{
   middleware.Next(context)
}
`

Of course, this would be very bad practice in any production code.

**Comment by jinzhu** [comment]
@sredXNY could you debug that in the `crsf_check` middleware? I am not sure what happened, but I think the header was dropped or so.

**Comment by tanphat198** [comment]
@sredxny hey , can i know how you solve this problem because i got the same situation with you now

**Comment by sredxny** [comment]
@tanphat198 this was a misconfiguration of the HTTP server, for some reason it was dropping the `Referer` header
